doc/compressed_transactions.md

```diff
@@ -0,0 +1,49 @@
+# Compressed Transaction Schema
+
+## 1. Abstraction
+With this transaction compression schema we reduce the size of all transactions by at least 15-20% and up to 40% with signature recovery. The main current uses for this will be steganography or other low bandwidth channels for passing transaction data.
+
+## 2. Methods
+The four main methods to achieve a lower transactions size are:
+1. Adding a few bytes of data before the transaction and each of its inputs and outputs to determine the structure of the following data.
+2. Removing all excess white space and fillers with the use of Variable Integers.
+3. The use of compressed signatures and public key recovery upon decompression
+4. Replacing the 32 byte Txid with a block height and index for the transaction instead.
+
+## 2. Schema
+
+| Name                           | Value     | Description |
+| -------------------------------|-----------|-------------|
+| Control VarInt(CV)             | 1-2 Bytes | A Variable Integer containing information on the structure of the transaction. |
+| CV->Version                    | 2 Bits    | If the Version is less then 4 the Version is encoded directly, otherwise 0b00 signals the existence  of a future VarInt for the Version. |
+| CV->Input Count                | 2 Bits    | If the Input Count is less then 4 the Input Count is encoded directly, otherwise 0b00 signals the existence of a future VarInt for the Input Count. |
+| CV->Output Count               | 2 Bits    | If the Output Count is less then 4 the Output Count is encoded directly, otherwise 0b00 signals the existence of a future VarInt for the Output Count. |
+| CV->LockTime                   | 2 Bits    | 0x00 if the LockTime is zero, 0x01 if the LockTime has been shortened to the sixteen least significant bits, 0x02 if is not been shortened and signals the existence of a future VarInt for the LockTime. |
+| CV->Coinbase                   | 1 Bits    | Boolean to determine if this transaction is a Coinbase Transaction. |
```

Comment:
In 61504d1a4d7ea545b1880686e426b1acae7d26f0:

I suspect we can drop this bit entirely.

Reply:
We can use block index 0 for this. I get that the block index is at least a byte wide, so by having a bit we can sometimes avoid encoding it...but I think coinbase transactions are a "fallback" case. It's hard to think when somebody would ever even broadcast a coinbase transaction except inside a block.

Reply:
Fixed, Thanks!